waku/v2/node/wakunode2.nim

```diff
@@ -648,195 +639,252 @@ proc connectToNodes*(n: WakuNode, nodes: seq[PeerInfo]) {.async.} =
-        waku_node_errors.inc(labelValues = ["init_store_failure"])
-      else:
-        store = res.value
+      if conf.filternode != "":
+        setFilterPeer(node, conf.filternode)
 
-    mountStore(node, store, conf.persistMessages)
+  # 4/6 Start node and mounted protocols
+  proc startNode*(node: WakuNode, conf: WakuNodeConf) =
+    ## Start a configured node and all mounted protocols.
+    ## Resume history, connect to static nodes and start
+    ## keep-alive, if configured.
 
-    if conf.storenode != "":
-      setStorePeer(node, conf.storenode)
+    # Start Waku v2 node
+    waitFor node.start()
+    
+    # Resume historical messages, this has to be called after the node has been started
+    if conf.store and conf.persistMessages:
+      waitFor node.resume()
     
+    # Connect to configured static nodes
+    if conf.staticnodes.len > 0:
+      waitFor connectToNodes(node, conf.staticnodes)
+
+    # Start keepalive, if enabled
+    if conf.keepAlive:
+      node.startKeepalive()
+
+  # 5/6 Start monitoring tools and external interfaces
+  proc startExternal*(node: WakuNode, conf: WakuNodeConf) =
+    ## Start configured external interfaces and monitoring tools
+    ## on a Waku v2 node, including the RPC API and metrics
+    ## monitoring ports.
+    
+    if conf.rpc:
+      startRpc(node, conf.rpcAddress, Port(conf.rpcPort + conf.portsShift), conf)
+
+    if conf.metricsLogging:
+      startMetricsLog()
 
-  # Relay setup
-  mountRelay(node,
-             conf.topics.split(" "),
-             rlnRelayEnabled = conf.rlnRelay,
-             relayMessages = conf.relay) # Indicates if node is capable to relay messages
+    if conf.metricsServer:
+      startMetricsServer(conf.metricsServerAddress,
+        Port(conf.metricsServerPort + conf.portsShift))
   
-  # Keepalive mounted on all nodes
-  mountLibp2pPing(node)
+  let
+    conf = WakuNodeConf.load()
   
-  # Resume historical messages, this has to be called after the relay setup           
-  if conf.store and conf.persistMessages:
-    waitFor node.resume()
+  var
+    node: WakuNode  # This is the node we're going to setup using the conf
 
-  if conf.staticnodes.len > 0:
-    waitFor connectToNodes(node, conf.staticnodes)
+  ##############
+  # Node setup #
+  ##############
 
-  # NOTE Must be mounted after relay
-  if (conf.lightpushnode != "") or (conf.lightpush):
-    mountLightPush(node)
+  debug "1/6 Setting up storage"
 
-    if conf.lightpushnode != "":
-      setLightPushPeer(node, conf.lightpushnode)
-  
-  # Filter setup. NOTE Must be mounted after relay
-  if (conf.filternode != "") or (conf.filter):
-    mountFilter(node)
+  let (pStorage, mStorage) = setupStorage(conf)
 
-    if conf.filternode != "":
-      setFilterPeer(node, conf.filternode)
+  debug "2/6 Initialising node"
 
-  if conf.rpc:
-    startRpc(node, conf.rpcAddress, Port(conf.rpcPort + conf.portsShift), conf)
+  node = initialiseNode(conf, pStorage).tryGet()
```

Comment:
I'd call this initNode, also initialise/initialize easy to get wrong (I usually prefer initialize, AmE)

Reply:
😄 My inner South African schooling system (BrE) is somewhat pained, but I'm all for consistency! Changed to `init` in 4cca119 and _standardized_ (😰) spelling to AmE in comments.